my-app/src/app/components/header/header.component.ts

```diff
@@ -1,14 +1,25 @@
-import {ChangeDetectionStrategy, Component} from '@angular/core';
+import {ChangeDetectionStrategy, Component, OnInit} from '@angular/core';
+
+import {AuthService} from "@core/services/auth.service";
 
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class HeaderComponent {
+export class HeaderComponent implements OnInit {
+  public isAuthenticated: boolean = false;
+
+  constructor(private authService: AuthService) {
+  }
 
-  constructor() {
+  ngOnInit() {
+    this.isAuthenticated = this.authService.IsAuthenticated;
```

Comment:
Should `isAuthenticated` be updated when it's updated in the service? It would be much easier to have

```
public isAuthenticated$: Observable<boolean> = this.authService.IsAuthenticated$;
```
😉

Reply:
Fixed, thank you